### ISSUE TITLE
"sh: 1: vite: not found" error from wails dev with Svelte template

### ISSUE THREAD

**Issue by ianmjones** [wailsapp/wails#1611]
### Description

With wails v2.0.0-beta.39.2, initialize a new project using the svelte template, then run `wails dev`, the project fails to run as it is unable to find `vite`:

```
ian@ubuntu-jammy-qemu:~/Projects/scratch$ wails init -t svelte -n wailsv2b39.2
Wails CLI v2.0.0-beta.39.2

Initialising Project 'wailsv2b39.2'
-----------------------------------

go: downloading github.com/wailsapp/wails/v2 v2.0.0-beta.39.2
Project Name:      wailsv2b39.2
Project Directory: /home/ian/Projects/scratch/wailsv2b39.2
Project Template:  svelte
Template Support:  https://wails.io

Initialised project 'wailsv2b39.2' in 2.966s.


If Wails is useful to you or your company, please consider sponsoring the project:
https://github.com/sponsors/leaanthony

ian@ubuntu-jammy-qemu:~/Projects/scratch$ cd wailsv2b39.2/
ian@ubuntu-jammy-qemu:~/Projects/scratch/wailsv2b39.2$ wails dev
Wails CLI v2.0.0-beta.39.2

Executing: go mod tidy -compat=1.17
Executing: wails generate module

> frontend@0.0.0 dev
> vite

sh: 1: vite: not found


ERROR: failed to find Vite server URL

If Wails is useful to you or your company, please consider sponsoring the project:
https://github.com/sponsors/leaanthony

ian@ubuntu-jammy-qemu:~/Projects/scratch/wailsv2b39.2$
```

### To Reproduce

1. Using wails v2.0.0-beta.39.2
2. wails init -t svelte -n wailsv2b39.2
3. cd wailsv2b39.2
4. wails dev

### Expected behaviour

A running stock template app.

### System Details

The following is from Ubuntu 22.04, using the latest `master` branch (which only has an additional fix for the plain template), but I also saw it on stock v2.0.0-beta.39.2, including on NixOS:

```shell
Wails CLI v2.0.0-beta.39.2

Scanning system - Please wait (this may take a long time)...Done.

System
------
OS:		Ubuntu
Version: 	22.04
ID:		ubuntu
Go Version:	go1.18.1
Platform:	linux
Architecture:	amd64

Wails
------
Version: 		v2.0.0-beta.39.2
Revision:		1f62b5773d7a87b8a3ce04c76b0aaaab32fda691
Modified:		false
Package Manager: 	apt

Dependency	Package Name		Status		Version
----------	------------		------		-------
*docker		docker.io		Available	20.10.12-0ubuntu4
gcc		build-essential		Installed	12.9ubuntu3
libgtk-3	libgtk-3-dev		Installed	3.24.33-1ubuntu2
libwebkit	libwebkit2gtk-4.0-dev	Installed	2.36.4-0ubuntu0.22.04.1
npm		npm			Installed	8.5.5
*nsis		nsis			Available	3.08-2
pkg-config	pkg-config		Installed	0.29.2-1ubuntu3

* - Optional Dependency

Diagnosis
---------
Your system is ready for Wails development!
Optional package(s) installation details: 
  - docker: sudo apt install docker.io
  - nsis: sudo apt install nsis
```

**Comment by stffabi** [comment]
Hi @ianmjones thanks for reporting.

Yeah, it seems like it doesn't do an npm install before trying to start the dev watcher. Unfortunately a regression introduced with the auto port discovery with PR #1547.

**Comment by ianmjones** [comment]
Interestingly, an existing project (snippetpixiegui) that also uses svelte, is working ok.

Existing project's wails.json:

```json
{
  "name": "snippetpixiegui",
  "outputfilename": "snippetpixiegui",
  "frontend:install": "npm install",
  "frontend:build": "npm run build",
  "frontend:dev:watcher": "npm run dev",
  "frontend:dev:serverUrl": "http://localhost:3000",
  "author": {
    "name": "Ian M. Jones",
    "email": "ian@ianmjones.com"
  }
}
```

New stock wails.json:

```json
{
  "name": "wailsv2b39.2",
  "outputfilename": "wailsv2b39.2",
  "frontend:install": "npm install",
  "frontend:build": "npm run build",
  "frontend:dev:watcher": "npm run dev",
  "frontend:dev:serverUrl": "auto",
  "author": {
    "name": "Ian M. Jones",
    "email": "ian@ianmjones.com"
  }
}
```

The only real difference is that `frontend:dev:serverUrl` has switched from a value of "http://localhost:3000" to "auto", and the respective `package.json` files only really differ in the version of `vite`, going from "^2.9.5" to "^2.9.9", which should be inconsequential.

When I update the new project's `frontend:dev:serverUrl` to use ye olde "http://localhost:3000", it gets a little further, but still in spite of not having found `vite`:

```
$ wails dev
Wails CLI v2.0.0-beta.39.2

Executing: go mod tidy -compat=1.17
Executing: /nix/store/s71knwxx9ldzkg20cw3zkq40z0qrdmsc-wails-2.0.0-beta.39.2/bin/wails generate module
Running frontend DevWatcher command: 'npm run dev'
Building application for development...
  - Installing frontend dependencies:
> frontend@0.0.0 dev
> vite

sh: line 1: vite: command not found
Error from DevWatcher 'npm run dev': exit status 127
DevWatcher command exited!

    npm ERR! code ERESOLVE
    npm ERR! ERESOLVE unable to resolve dependency tree
    npm ERR!
    npm ERR! While resolving: frontend@0.0.0
    npm ERR! Found: vite@2.9.14
    npm ERR! node_modules/vite
    npm ERR!   dev vite@"^2.9.9" from the root project
    npm ERR!
    npm ERR! Could not resolve dependency:
    npm ERR! peer vite@"^3.0.0" from @sveltejs/vite-plugin-svelte@1.0.1
    npm ERR! node_modules/@sveltejs/vite-plugin-svelte
    npm ERR!   dev @sveltejs/vite-plugin-svelte@"^1.0.0-next.30" from the root project
    npm ERR!
    npm ERR! Fix the upstream dependency conflict, or retry
    npm ERR! this command with --force, or --legacy-peer-deps
    npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
    npm ERR!
    npm ERR! See /home/ian/.npm/eresolve-report.txt for a full report.

    npm ERR! A complete log of this run can be found in:
    npm ERR!     /home/ian/.npm/_logs/2022-07-20T07_14_23_748Z-debug-0.log


Build error - exit status 1
No version running, build will be retriggered as soon as changes have been detected
Watching (sub)/directory: /home/ian/Projects/sr.ht/ianmjones/snippetpixie/wailsv2b39.2
Using DevServer URL: http://localhost:34115
Using Frontend DevServer URL: http://localhost:3000
Using reload debounce setting of 100 milliseconds


To develop in the browser, navigate to: http://localhost:34115
```

**Comment by stffabi** [comment]
Relates to #1182

**Comment by ianmjones** [comment]
> Hi @ianmjones thanks for reporting.
> 
> Yeah, it seems like it doesn't do an npm install before trying to start the dev watcher. Unfortunately a regression introduced with the auto port discovery with PR #1547.

Ha, comments crossed in the ether.

Was going to mention that there seems to be a missing call to install the npm dependencies.

> Relates to #1182

Ah, I see.

**Comment by leaanthony** [comment]
I noticed this when I did a fresh install with no global vite so added an installation step before the dev watcher is run. I'll need to check whether it was pushed or not 😔

**Comment by ianmjones** [comment]
Just to confirm, I've run `make clean-all` on my existing project so that it gets `node_modules` etc removed, and sure enough, it now will not run with `wails dev` because there's no step to install the frontend dependencies.

**Comment by leaanthony** [comment]
I'll fix it tonight and get .3 out

**Comment by leaanthony** [comment]
@ianmjones fix in the linked PR. Let us know! 👍

**Comment by ianmjones** [comment]
@leaanthony, hmm, something's not quite right with a checkout of `master` (now that the PR is merged) and replace in go.mod pointing to the checkout:

```
ian@ubuntu-jammy-qemu:~/Projects/scratch/wailsv2b39.3$ wails dev
Wails CLI v2.0.0-beta.39.2

Executing: go mod tidy -compat=1.17
Executing: wails generate module
Installing frontend dependencies...


ERROR: io: read/write on closed pipe

If Wails is useful to you or your company, please consider sponsoring the project:
https://github.com/sponsors/leaanthony

ian@ubuntu-jammy-qemu:~/Projects/scratch/wailsv2b39.3$ wails doctor
Wails CLI v2.0.0-beta.39.2

Scanning system - Please wait (this may take a long time)...Done.

System
------
OS:		Ubuntu
Version: 	22.04
ID:		ubuntu
Go Version:	go1.18.1
Platform:	linux
Architecture:	amd64

Wails
------
Version: 		v2.0.0-beta.39.2
Revision:		c9a7d3f5207524b3242889cf9d1c994b6920171e
Modified:		false
Package Manager: 	apt

Dependency	Package Name		Status		Version
----------	------------		------		-------
*docker		docker.io		Available	20.10.12-0ubuntu4
gcc		build-essential		Installed	12.9ubuntu3
libgtk-3	libgtk-3-dev		Installed	3.24.33-1ubuntu2
libwebkit	libwebkit2gtk-4.0-dev	Installed	2.36.4-0ubuntu0.22.04.1
npm		npm			Installed	8.5.5
*nsis		nsis			Available	3.08-2
pkg-config	pkg-config		Installed	0.29.2-1ubuntu3

* - Optional Dependency

Diagnosis
---------
Your system is ready for Wails development!
Optional package(s) installation details: 
  - docker: sudo apt install docker.io
  - nsis: sudo apt install nsis



If Wails is useful to you or your company, please consider sponsoring the project:
https://github.com/sponsors/leaanthony

ian@ubuntu-jammy-qemu:~/Projects/scratch/wailsv2b39.3$
```

That was in a project newly created with `wails init -t svelte -n wailsv2b39.3`.

**Comment by ianmjones** [comment]
This may explain things?

```
ian@ubuntu-jammy-qemu:~/Projects/scratch/wailsv2b39.3$ wails build
Wails CLI v2.0.0-beta.39.2

App Type: 		desktop
Platforms: 		linux/amd64
Compiler: 		/usr/bin/go
Build Mode: 		Production
Skip Frontend: 		false
Compress: 		false
Package: 		true
Clean Build Dir: 	false
LDFlags: 		""
Tags: 			[]
Race Detector: 		false

Building target: linux/amd64
----------------------------
  - Installing frontend dependencies:     
    npm ERR! code ERESOLVE
    npm ERR! ERESOLVE unable to resolve dependency tree
    npm ERR! 
    npm ERR! While resolving: frontend@0.0.0
    npm ERR! Found: vite@2.9.14
    npm ERR! node_modules/vite
    npm ERR!   dev vite@"^2.9.9" from the root project
    npm ERR! 
    npm ERR! Could not resolve dependency:
    npm ERR! peer vite@"^3.0.0" from @sveltejs/vite-plugin-svelte@1.0.1
    npm ERR! node_modules/@sveltejs/vite-plugin-svelte
    npm ERR!   dev @sveltejs/vite-plugin-svelte@"^1.0.0-next.30" from the root project
    npm ERR! 
    npm ERR! Fix the upstream dependency conflict, or retry
    npm ERR! this command with --force, or --legacy-peer-deps
    npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
    npm ERR! 
    npm ERR! See /home/ian/.npm/eresolve-report.txt for a full report.
    
    npm ERR! A complete log of this run can be found in:
    npm ERR!     /home/ian/.npm/_logs/2022-07-20T12_05_18_464Z-debug-0.log
    
Error: exit status 1


ERROR: exit status 1

If Wails is useful to you or your company, please consider sponsoring the project:
https://github.com/sponsors/leaanthony
```

**Comment by leaanthony** [comment]
You using vite3?

**Comment by leaanthony** [comment]
Oh I see what's happening. Yeah, the vite-plugin-svelte project mandated vite3 which broke stuff 😠 There's a PR in progress for that #1580

**Comment by ianmjones** [comment]
> Oh I see what's happening. Yeah, the vite-plugin-svelte project mandated vite3 which broke stuff angry There's a PR in progress for that #1580
> 
> Does this fix the initial issue?

I think that'll require merging master into the PR's branch to bring it up to date before I could give it a go. I'll have a look at that this evening if you don't beat me to it.

**Comment by ianmjones** [comment]
Yup, PR #1580 fixed things up for me once I merged master into it and resolved the conflicts.

**Comment by ianmjones** [comment]
Closing as verified fixed in v2.0.0-beta.42.
